### DISCUSSION TITLE
Current and latest version information should be available to the end user

### DISCUSSION THREAD

**Discussion by andreszs** [comfyanonymous/ComfyUI#1606] [Ideas]
In the GUI, there's no dialog where we can see the currently installed version, or the latest available version for that matter. I struggled for hours trying to install an extension which resulted in Missing Nodes, only to find out that all that was needed was updating ComfyUI. I used the Manager extension to update it and it solved the problem.

I still don't know what version I'm using nor how to check for updates.

**Comment by brbbbq** [comment]
I second this. A developer asked me what version of ComfyUI I'm using, and I'm scouring the commandline arguments, command window log, github, and the repo website - I can't find a single piece of info that looks like a version number anywhere.

I'm not a coder/developer, so I don't know what's the standard, but seems odd that it's impossible to find a basic piece of information.

Also, the only way to check for updates is to install and use the ComfyUI Manager, but I thought there would at least be a check and confirmation first. Nope, once you click the update button it runs a pip install and you're already commited before having a chance to review anything.

Now I'm having a different set of issues than before because I'm running a new version that broke something else. Very messy.

**Reply by andreszs** [reply]
You are right about the update process with the Manager extension, it feels like a leap into the void. The last time, it took so long that I was already searching for the external drive to restore the comfyui backup. It seems to have succeeded somehow, but I haven't tested all my extensions yet.

**Comment by dadadies** [comment]
Has this version issue ever been addressed. I never know what version of comfyui i have and if i was able update at all. I am using the portable so i use the update.bat file but i never know if it ever updates. I would expect at least a message saying, already up to date, or updated.

**Comment by andreszs** [comment]
No it hasn't, the developers seem to believe that's absolutely irrelevant for users to know which version number they are using. We simply have **no clue** on how to check our ComfyUI version at this time.

**Comment by vcsoc** [comment]
Using the Command Line, navigate to your ComfyUI directory and execute the command 'git log -1'. 
This command will show you the latest commit, which can help you identify the version based on the latest changes in the repository.

**Reply by ltdrdata** [reply]
![image](https://github.com/user-attachments/assets/89719b7b-e22f-4bf2-8361-42af0b7c4f03)

ComfyUI has now started using semantic versioning based on tags.
In the future, ComfyUI-Manager also plans to introduce version management functionality for these versions.
